### PR TITLE
chore: add file type reference rule

### PR DESCRIPTION
**Pull request by lennessyy**
## Describe the Change

This PR adds a Vale rule that standardize how to refer to file types. I am using the "XXX file" as the match rule specifically to reduce noise. 

## Review Changes

💻 [Add Preview URL]()

🎫 [Jira Ticket]()

### CODE REVIEW

File: vale/styles/spectrocloud/file-type.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+extends: substitution
+message: "Incorrect file type reference. Use '%s'."
+level: error
+ignorecase: false
+swap:
+  csv file: CSV file
+  json file: JSON file
+  png file: PNG file
+  svg file: SVG file
+  TAR file: tar file
```
Comment (addetz) [inline vale/styles/spectrocloud/file-type.yml:10]: Why is this not in ALL CAPS?

File: vale/styles/spectrocloud/file-type.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  tf file: Terraform file
+  txt file: text file
+  yaml file: YAML file
+  ZIP file: zip file
+  ZST file: zst file
+  TGZ file: tgz file
```
Comment (addetz) [inline vale/styles/spectrocloud/file-type.yml:16]: These seem bizarre to me when they're not in ALL CAPS.

File: vale/styles/spectrocloud/file-type.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+extends: substitution
+message: "Incorrect file type reference. Use '%s'."
+level: error
+ignorecase: false
+swap:
+  csv file: CSV file
+  json file: JSON file
```
Comment (addetz) [inline vale/styles/spectrocloud/file-type.yml:7]: Is there an extra space here?

**Review by addetz — COMMENTED** [review]
I question some of these swaps. To me, `TAR`, `ZST` and `TGZ` make no sense if they are not in all caps.

File: vale/styles/spectrocloud/file-type.yml (hunk 2 above)
Comment (karl-cardenas-coding) [reply vale/styles/spectrocloud/file-type.yml:16]: I agree

**Review by karl-cardenas-coding — COMMENTED** [review]
I agree with Adelina's feedback

**Review by karl-cardenas-coding — APPROVED** [review]

**Pull request merged**